www/install/php/Update-20.04.0-beta.1.php

```diff
@@ -90,6 +89,111 @@
     throw new \PDOException($e);
 }
 
+// Generate a new config file for the gorgone daemon module
+try {
+    //set default values for new mandatory parameters used by gorgone
+    $pattern = [
+        '/--CENTREON_SPOOL--/',
+        '/--HTTPSERVERADDRESS--/',
+        '/--HTTPSERVERPORT--/',
+        '/--SSLMODE--/',
+        '/--CENTREON_TRAPDIR--/',
+    ];
+
+    $userValues = [
+        '/var/spool/centreon',
+        '0.0.0.0',
+        '8085',
+        'false',
+        '/etc/snmp/centreon_traps',
+    ];
+
+    $instanceMode = "";
+
+    $fileToOpen = _CENTREON_ETC_ . './conf.pm';
+
+    // check if conf.pm exists and is not empty
+    if (!file_exists($fileToOpen) || 0 === filesize($fileToOpen)) {
+        throw new InvalidArgumentException('Missing \'conf.pm\' file');
+    }
+    // getting user's data from the file
+    $file = fopen($fileToOpen, 'r');
+
+    while (!feof($file)) {
+        // removing useless characters
+        $line = str_replace(
+            [" ", ",", ";", "{", "}", "\"", "'", ">", "\n", "$"],
+            "",
+            fgets($file)
+        );
+        // checking for not commented string length
+        if (strpos($line, "#") === false && strlen($line) > 1) {
+            $line = explode("=", $line);
+            list($currentLine, $currentValue) = $line;
+
+            if (strlen($currentValue) > 1) {
+                switch ($currentLine) {
```

Comment:
More simple: use an array to store the pattern values.
$patterns = ['instance_mode' => '/--INSTANCEMODE--/', ...]

Reply:
The "instance_mode" isn't a key I need to keep.
The key need to be an integer matching the one of the second array to be able to use the preg_replace later